extra/redisprom/collector.go

```diff
@@ -0,0 +1,81 @@
+package redisprom
+
+import (
+	"github.com/go-redis/redis/v8"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type poolStatsCollector struct {
+	universeClient redis.UniversalClient
+
+	hits     *prometheus.Desc
+	misses   *prometheus.Desc
+	timeouts *prometheus.Desc
+
+	totalConns *prometheus.Desc
+	idleConns  *prometheus.Desc
+	staleConns *prometheus.Desc
+}
+
+// NewPoolStatsCollector returns a collector that exports pool metrics about the given redis.UniversalClient.
+func NewPoolStatsCollector(universeClient redis.UniversalClient, name string) prometheus.Collector {
```

Comment:
Probably just `redisprom.NewCollector`? In theory, this collector can gather other stats as well...